RHDEVDOCS-3358 - new content for dex and keycloak (#39282)
openshift/openshift-docs: modules/gitops-creating-a-new-client-using-keycloak.adoc

[id="creating-a-new-client-in-keycloak_{context}"]
= Creating a new client in Keycloak

.Procedure

> . Create a new Argo CD Instance in the `argocd` namespace

Reviewer: This is not correct. It should look something like below.

1. Configure keycloak.
-> Edit the Argo CD CR and update `.sso.spec.provider: keycloak`

> Keycloak acts as an identity broker between Argo CD and OpenShift; as a result, you can log in to Argo CD using the OpenShift credentials.

Reviewer 2: ```suggestion
Keycloak acts as an identity broker between Argo CD and OpenShift; as a result, you can log in to Argo CD using your OpenShift credentials.
```

+
[source,yaml]
----
kubectl create -n argocd -f examples/argocd-keycloak.yaml
----
. After creating the Argo CD instance, login into the keycloak console.
+
[source,yaml]
----

> kubectl -n argocd get route keycloak

Reviewer: please remove this

> . Dex is installed by default for all the Argo CD instances created by the operator. For Keycloak to work, first delete the dex configuration from the Argo CD CR. Remove the following and *Save* the CR:

Reviewer 2: ```suggestion
Dex is installed by default for all the Argo CD instances created by the Operator. For Keycloak to work, follow these steps:
. Delete the Dex configuration from the Argo CD CR by removing the following and saving the CR:
```
- The first sentence is not a step, and so shouldn't be in the procedure. Maybe use the first sentence as an intro to the procedure, as my edit shows.
- Per the OCP doc guidelines, `Operator` is always capitalized: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-capitalization
- `CR` should be spelled out on first use. See the "Spelled-out forms" section of the IBM SG.

        memory: 
      requests:
        cpu: 
        memory: 
----

> . Configure Keycloak by editing and updating the Argo CD `.spec.sso.provider: keycloak`.

Reviewer 2: ```suggestion
. Configure Keycloak by editing and updating the Argo CD `.spec.sso.provider: keycloak` as shown in this sample:
```

> . Get the Keycloak Pod Name which is stored as environment variables in the keycloak pod.

Reviewer: please move this under "Logging into Keycloak" section.

Also there is a modification.
Get the Keycloak credentials which are stored as environment variables in the keycloak pod.

----
  kubectl -n argocd get pods

  NAME                                         READY   STATUS             RESTARTS   AGE
  keycloak-1-2sjcl                                  1/1     Running            0          45m
----

> . Get the Keycloak username.

Reviewer: please move this under "Logging into Keycloak" section.

+
[source,yaml]
----
  kubectl -n argocd exec keycloak-1-2sjcl -- "env" | grep SSO_ADMIN_USERNAME

SSO_ADMIN_USERNAME=Cqid54Ih
----

> . Get the Keycloak password.

Reviewer: please move this under "Logging into Keycloak" section.

[id="creating-a-new-client-in-keycloak_{context}"]
= Creating a new client in Keycloak

.Procedure

> . Configure Keycloak.

Reviewer 3: For one step procedures, use a bullet point, not a number. See Mod. Doc. guidelines for procedures (I have to reference this all the time!) https://redhat-documentation.github.io/modular-docs/#creating-procedure-modules 

Also, usually in procedures, if there's a sub-step, it's providing more information, an alternate way of completing the action, or giving a note or heads up. What do you think about turning these actions into one line, or one action? Something like this:

Configure Keycloak by editing the Argo CD and updating `.sso.spec.provider: keycloak`.

[id="creating-a-new-client-in-keycloak_{context}"]
= Creating a new client in Keycloak

> Create a new client in Keycloak for secure and easy identity and access management. Keycloak acts as an Identity broker between ArgoCD and OpenShift, as a result you can login into Argo CD using the OpenShift credentials also.

Reviewer 3: minor grammatical changes. what do you think about this rewrite:

`Create a new client in Keycloak for security, easy identity, and access management. Keycloak acts as an identity broker between ArgoCD and OpenShift; as a result, you can login into Argo CD using the OpenShift credentials.` 

Note, I looked up `identify broker` in other RH documentation and they don't capitalize See this page for an example: https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/server_administration_guide/identity_broker

.Procedure

> . Configure Keycloak by editing and updating the Argo CD `.sso.spec.provider: keycloak`.

Reviewer 3: My bad for not noting this before. But for single step procedures, they want you to use a bullet point, not a number. That's the only thing I'd change in all these docs. They all look great!

> . Configure Keycloak by editing the Argo CD and updating `.sso.spec.provider: keycloak`.

Reviewer 3: I read this step this way,

Configure Keycloak by editing **and** updating the Argo CD  **with** `.sso.spec.provider: keycloak`.

How it's currently written makes me think you configure Keycloak by editing the Argo CD and you edit the Argo CD by updating it with this sso spec. Is that right?
